CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKIconsRenderable.java

```diff
@@ -956,381 +1460,627 @@ public int DestroyDescriptorPoolResources() {
-    }    
+    }  
+    
+    private void RebuildVertexFlagsStagingBuffers(final VisualAccess access, final int newVertexCount) {
+        try {
+            // Vertices are modified by the event thread
+            vertexFlagsStagingBufferLock.lock(); 
+            
+            // Destroy old staging buffer if it exists
+            if (cvkVertexFlagsStagingBuffer != null) {
+                cvkVertexFlagsStagingBuffer.Destroy();
+                cvkVertexFlagsStagingBuffer = null;
+            }       
+            
+            if (newVertexCount > 0) {
+                int vertexFlagsBufferSizeBytes = newVertexCount * FLAGS_STRIDE;
+                cvkVertexFlagsStagingBuffer = CVKBuffer.Create(cvkDevice, 
+                                                               vertexFlagsBufferSizeBytes, 
+                                                               VK_BUFFER_USAGE_TRANSFER_SRC_BIT,
+                                                               VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
+                cvkVertexFlagsStagingBuffer.DEBUGNAME = "CVKIconsRenderable.RebuildVertexFlagsStagingBuffers cvkVertexFlagsStagingBuffer";
+                
+                ByteBuffer pVertexFlagsMemory = cvkVertexFlagsStagingBuffer.StartWrite(0, vertexFlagsBufferSizeBytes);
+                for (int pos = 0; pos < newVertexCount; pos++) {
+                    SetVertexFlags(pos, 0, pVertexFlagsMemory, access);
+                }
+                cvkVertexFlagsStagingBuffer.EndWrite();
+                pVertexFlagsMemory = null; // now unmapped, do not use           
+            }
+        } finally {
+            vertexFlagsStagingBufferLock.unlock();
+        }         
+    }
+    
+    public CVKRenderableUpdateTask TaskRebuildVertexFlags(final VisualAccess access) {
+        //=== EXECUTED BY CALLING THREAD (VisualProcessor) ===//
+        final int newVertexCount = access.getVertexCount();
+//        CVKAssert(newVertexCount == vertexCount);
+        CVKLOGGER.info(String.format("TaskRebuildVertexFlags frame %d: %d verts", parent.GetFrameNumber(), access.getVertexCount()));
+        RebuildVertexFlagsStagingBuffers(access, newVertexCount);
+        
+        //=== EXECUTED BY RENDER THREAD (during CVKVisualProcessor.ProcessRenderTasks) ===//
+        return () -> {            
+            vertexFlagsBufferState = CVK_RESOURCE_NEEDS_REBUILD;
+        };     
+    }
+    
+    public CVKRenderableUpdateTask TaskUpdateVertexFlags(final VisualChange change, final VisualAccess access) {
+        //=== EXECUTED BY CALLING THREAD (VisualProcessor) ===//
+        // If this fires investigate why we didn't get a rebuild task first
+        CVKLOGGER.info(String.format("TaskUpdateVertexFlags frame %d: %d verts", parent.GetFrameNumber(), access.getVertexCount()));
+        final int newVertexCount = access.getVertexCount();
+//        CVKAssert(vertexCount != newVertexCount); //REMOVE AFTER TESTING
```

Comment:
Remove?

Reply:
I'm not ready yet, I don't know if we can get say a flags update which indicates n + x vertices when the last vertex rebuild only had n.  I need to understand this a bit better before cleaning up the code (messy code annoys so I know I won't ignore it).